### ISSUE TITLE
Installing mo2 for starfield breaks the game

### ISSUE THREAD

**Issue by scrubborn** [rockerbacon/modorganizer2-linux-installer#679]
Installing mo2 on a fresh copy of starfield with a fresh prefix causes game crashes for me once I hit the new game button. It loads for a moment and I hear the "thud" and it freezes. I could start a new game fine before installing this. Issue happens if I open the game with sfse or just normally.  
  
I'm currently on zorinos but I noticed the same issue on garuda as well. I have completely reinstalled the game into a fresh directory, and remade the prefix, and still have the issue.  
  
The game has functioned perfectly for me since launch so this has me quite confused.

**Comment by jSQrD-dev** [comment]
Having a similar issue since the latest update. I would load into a save and every planet is now Starfield branded. Verifying Game Files fixes it.

**Comment by tdtooke** [comment]
Starfield script extender has been updated again.  Don't know if that's relevant to your problem.  What you can do is go to "gamesinfo/starfield.sh" and change all instances of "sfse_0_2_7" to "sfse_0_2_8".  Then you want to change the dll to "sfse_1_12_30.dll" and build it.

**Comment by jSQrD-dev** [comment]
> Starfield script extender has been updated again. Don't know if that's relevant to your problem. What you can do is go to "gamesinfo/starfield.sh" and change all instances of "sfse_0_2_7" to "sfse_0_2_8". Then you want to change the dll to "sfse_1_12_30.dll" and build it.
> 
> edit: I recently uninstalled Starfield so I could do Gold Coast on ESO so I can't really check into it. I need a bigger hard drive!

Still broken. 

![Screenshot_20240611_050402](https://github.com/rockerbacon/modorganizer2-linux-installer/assets/39681015/be97acce-03f3-41c2-b4d1-03c3998e9804)

**Comment by Qrtrdane** [comment]
I'm on Linux Mint Debian Edition 6 (based on Debian 12 Bookworm), and am facing a similar issue as scrubborn - attempting to launch Starfield via MO2 (with or without SFSE) fails (and crashes) when starting a new game or loading an existing one.
Note that due to Proton 9 causing an "Assertion Failed" error (to which I've only found solutions that require a different distro), I'm using Proton Experimental instead.
If I run Starfield normally (i.e. remove the underscore from _Starfield.exe and launch via Steam), the game seems to work without issue.

**Comment by NaturalHate** [comment]
I just get a "couldn't receive exe version information" error when I try to launch the game through MO2.
What's the process of updating SFSE for MO2? It seems to use an outdated version, 0.2.6, while 0.2.9 seems to be the latest one.

**Comment by Qrtrdane** [comment]
> I just get a "couldn't receive exe version information" error when I try to launch the game through MO2. What's the process of updating SFSE for MO2? It seems to use an outdated version, 0.2.6, while 0.2.9 seems to be the latest one.
> 
> Edit: Manually dropping the new version in does not seem to do anything as it just overrides it with the old version. :/

To confirm, are you dropping the newer SFSE into Steam/steamapps/common/Starfield?
Also did you make sure you 1: used the latest mo2-installer, and 2: you have the latest version of Starfield (especially the executable)?

**Comment by Qrtrdane** [comment]
> Yes but when I try to launch the SFSE terminal window at the top still says 0.2.6.

Checked the terminal output for my SFSE, and that too seems to show 0.2.6 - presumably this is because no major changes have happened to the *loader*, with all the actual changes that increment the version of the *download* being relegated to the enclosed game-version-specific .dll file.

> I did notice that there's a 103,7 MB large _Starfield.exe and a 96,6 KB small Starfield.exe after installing MO2 though, which I don't remember if that was the case in my original installation.

Pretty sure it is - the Starfield.exe being there so launching via Steam brings you to MO2 instead, and _Starfield.exe being the actual game executable.

On that note, if you check your Starfield/Data/SFSE/sfse.ini, does it have the following?
```
[Loader]
RuntimeName=_Starfield.exe
```

**Comment by NaturalHate** [comment]
> On that note, if you check your Starfield/Data/SFSE/sfse.ini, does it have the following?

It did not actually create a SFSE folder, and that seems to be the issue.
I backed up my old one though and it has that line so I guess I never noticed that file before, or forgot about it.
I manually created the folder and copied the old sfse.ini over which did make me launch the game.

Though, this brings me to a question: Am I supposed to enable the .esm mods manually within the game's creation menu? Because I cannot enable them from within MO2. Maybe preferable with the hard to read fonts but I still feel I want a native mod manager, or r2modman to support Bethesda games. :/

**Comment by Qrtrdane** [comment]
> Though, this brings me to a question: Am I supposed to enable the .esm mods manually within the game's creation menu? Because I cannot enable them from within MO2.

Do you have the plugins.txt enabler mod? https://www.nexusmods.com/starfield/mods/4157

> Maybe preferable with the hard to read fonts

Have you tried setting "fontsmooth=rgb" for Starfield via winetricks/protontricks?

**Comment by NaturalHate** [comment]
> Do you have the plugins.txt enabler mod? https://www.nexusmods.com/starfield/mods/4157

No. The mod page said the mod wasn't necessary anymore so I left it.

> Have you tried setting "fontsmooth=rgb" for Starfield via winetricks/protontricks?

I meant the font issue in MO2, not the game.

**Comment by Qrtrdane** [comment]
> No. The mod page said the mod wasn't necessary anymore so I left it.

Despite this, the version of MO2 currently available via the installer doesn't seem to allow enabling them unless you have said mod installed.

Some of the newer builds of MO2 (available on their discord) have this fixed, but not sure how you'd use those with the installer, so...

> I meant the font issue in MO2, not the game.

So did I. Since MO2 is launched _as_ Starfield, or rather using _Starfield's wine/proton context_, adjusting the settings _for_ Starfield will also affect MO2.

**Comment by NaturalHate** [comment]
> So did I. Since MO2 is launched _as_ Starfield, or rather using _Starfield's wine/proton context_, adjusting the settings _for_ Starfield will also affect MO2.

I'll try later because I think I have to re-install MO2 again anyway now, because I did a file check in Steam since I somehow miss the Akila City dirt roads and that seem to have botched the MO2 integration... Without fixing my issue of course. Or maybe I'll just uninstall again. Seems like this is all cursed.

**Comment by NaturalHate** [comment]
Okay so the mod does indeed enables the functionality within MO2. However, MO2 cannot enable the four new .esm files from Bethesda. Whenever I start the game they get disabled again.

But it turns out the game does simply not render any natural ground textures around a certain radius around me anymore for some reason, so I guess I forfeit my Starfield endeavor for good, since absolutely everything seems to be so broken with this game.

**Comment by Xeraster** [comment]
I can't get this to work on Skyrim Special Edition. Proton version 8 and 9 it does the same thing. I go through the installer, it finishes and then when I run the game in steam, the game crashes immediately without even showing so much as a menu or a loading screen. I've tried reinstalling this 5 different times and i've deleted my skyrim files and redownloaded twice.

**Comment by tdtooke** [comment]
Just for general info, everything works fine still if you do your mods manually on Starfield, sfse, all the other mods (at least the ones I use).  I suspect Creations managing plugins.txt is the problem.  I haven't tried MO2 with the latest updates but I suspect you may be able to manually enable those new esms inside of Creations with the load order thingee.  Not 100% sure since I'm not currently using MO2 for this.

**Comment by Qrtrdane** [comment]
> When I manually install the latest SFSE version for the latest version of the game, and I try to launch it through MO2, it complains that it cannot find the SFSE dll for 1.12.32, the old version.

Did you make sure you had the newest version of the actual Starfield.exe?

**Comment by tdtooke** [comment]
Not sure if this is the problem but if you manually install sfse you need "bash -c 'exec "${@/Starfield.exe/sfse_loader.exe}"' -- %command%" in the launch options.

**Comment by Qrtrdane** [comment]
> Not sure if this is the problem but if you manually install sfse you need "bash -c 'exec "${@/Starfield.exe/sfse_loader.exe}"' -- %command%" in the launch options.

Did you mean "when manually installing SFSE for use without MO2"? 'Cos I'm pretty sure NaturalHate was just doing a manual install to replace the SFSE installed by MO2-linux-installer.

**Comment by tdtooke** [comment]
Yeah, that's how I use sfse without MO2.

**Comment by NaturalHate** [comment]
> Did you make sure you had the newest version of the actual Starfield.exe?

Unless Steam did not download it with the update..? Or is MO2 interfering with this somehow?

>Not sure if this is the problem but if you manually install sfse you need "bash -c 'exec "${@/Starfield.exe/sfse_loader.exe}"' -- %command%" in the launch options.

I don't want to use SFSE without MO2 though? How else am I supposed to update SFSE if not manually?

**Comment by Qrtrdane** [comment]
> Unless Steam did not download it with the update..? Or is MO2 interfering with this somehow?

Yeah. You'll basically want to delete the actual one (_Starfield.exe), rename MO2's version (Starfield.exe) to something else, and verify integrity of game files. This should get you the up-to-date Starfield executable (which you then rename to _Starfield.exe, and rename the one from MO2 back to Starfield.exe).

If this doesn't work, you'll probably want to run the MO2 installer again.

**Comment by NaturalHate** [comment]
> If this doesn't work, you'll probably want to run the MO2 installer again.

You have to either way because the file integrity check removes that hook that this installer does to launch MO2 using the Play button in Steam. Which is also a bit of a trouble because there's been some issues with the stable release versions of Protontricks on most repos & flathub recently. 
Anyway, if that's the way to update the game and SFSE properly then that's really convoluted and confusing and should at least be mentioned on the main page.

**Comment by Qrtrdane** [comment]
> You have to either way because the file integrity check removes that hook that this installer does to launch MO2 using the Play button in Steam.

...Isn't that just from the installer's custom Starfield.exe replacing the original (which gets renamed _Starfield.exe)?

**Comment by Qrtrdane** [comment]
> If it renames the original executable for their own launcher then they have to make sure that it gets properly updated when the game updates.

Pretty sure that's from this unofficial installer rather than anything to do with MO2 proper, most likely so the whole shebang is launched with the correct wine/proton (and their respective settings) - which (AFAIK) isn't what you'd get from just installing the two separately and then launching MO2.

...Speaking of, I wonder if installing the two via Bottles (in the same "bottle") would work any better...

**Comment by NaturalHate** [comment]
> ...Speaking of, I wonder if installing the two via Bottles (in the same "bottle") would work any better...

I couldn't figure out bottles, too confusing and unintuitive for me. Even Lutris was easier and that already has an awful UI & UX. I also am not a fan of needing several hundred MB dependencies for it because of Gnome. We need a proper native mod manager for Bethesda games that does not need all those workarounds that may or may not work and are all a potential weak point in breaking upon game, software or system updates.

**Comment by Qrtrdane** [comment]
> We need a proper native mod manager for Bethesda games that does not need all those workarounds that may or may not work and are all a potential weak point in breaking upon game, software or system updates.

In the meantime, we have this installer - and also [this guide that I just stumbled upon](https://www.youtube.com/watch?v=B8-FUcGODcI) (it's for Skyrim, but seems like it'd work just as well for Starfield?).

**Comment by LinuxfarmerHH** [comment]
Did the setup in Zorin-OS for Starfield. No SFSE used, but few CK mods are installed. After install Steam opens mo2, the CK mods can be seen. Starting Starfield from mo2 results to a black Starfield start screen with CTD.

Steam file repair destroy mo2 but repairs Starfield for me.

**Comment by Canageek** [comment]
I have a very similar issue to the original reporter: Starfeild was working fine for me. I've done the introduction up until you find the weird artifact. I've not installed any script extenders or creation club mods, nothing. I follow the MO2 installer instructions, and it reports success. However, when I press the Play button on steam, nothing happens. After a bit it just goes back to normal without Starfield ever launching. I'm on Fedora's KDE version if that matters. 

I built this PC just to play Starfield, so I'm hoping this gets fixed or someone can help me.

**Comment by Dark-Thoughts** [comment]
Heads up: The latest Proton Experimental update that just came out seems to break this. Switch to a stable or GE build for now.

**Comment by Qrtrdane** [comment]
> Heads up: The latest Proton Experimental update that just came out seems to break this. Switch to a stable or GE build for now.

Break what, exactly? Prior to said update, I could launch Starfield by itself just fine, but not via MO2 (at least using this installer).

And yes, I had also tried Proton 9 (as was recommended by the installer instructions), but that just prevents the game from working at all due to an assert failure (as mentioned in my first post on this issue).

**Comment by Canageek** [comment]
Update: My problem turned out to be a manifestation of Bug 656: https://github.com/rockerbacon/modorganizer2-linux-installer/issues/656

**Comment by tdtooke** [comment]
I just saw something get released upstream that'll probably fix whatever is happening for you guys.  It won't be long now.

**Comment by Qrtrdane** [comment]
> I just saw something get released upstream that'll probably fix whatever is happening for you guys. It won't be long now.

The new version of MO2, or...?

**Comment by tdtooke** [comment]
MO2.  My theory is the problem you guys have is related to creations being there.  I think I read somewhere that some mods aren't working.  Not really sure, when I came back I just did everything manually since I was doing that for a good while before MO2 came out supporting Starfield.  The link is here for the new release: https://github.com/ModOrganizer2/modorganizer/releases
If you wanted to you could edit this to use that source and then you'd need to edit the Starfield script extender part to get the latest for it and it should in theory work.  In theory...

**Comment by hughesjr** [comment]
What version of MO2 are you running.  I now have version 2.5.2 running on nobara.  I also replaced game_starfield.dll with the new one in the MO2 discord 'dev-builds' channel to fix SFBGS004.esm from being properly managed.  The only error I am not getting has to do with **QtWindows::DpiAwareness::PerMonitorVersion2**  and that make MO2 look bad (fonts shimmer, etc), but everything works .. mods are installed.

**Comment by hughesjr** [comment]
> > Do you have the plugins.txt enabler mod? https://www.nexusmods.com/starfield/mods/4157
> 
> No. The mod page said the mod wasn't necessary anymore so I left it.
> 
> > Have you tried setting "fontsmooth=rgb" for Starfield via winetricks/protontricks?
> 
> I meant the font issue in MO2, not the game.

The font issue in MO2, I think is DpiAwareness issue .. it seems wine 9.10 and wine 9.11 both have DPI Awareness changes.  Hopefully this will fix the fonts in MO2 when it goes into the distros.  Nobara currently has wine 9.9-14, so maybe I'll get an update soon.

**Comment by Dark-Thoughts** [comment]
> What version of MO2 are you running.

Still 2.5.0, as per the latest 5.0.3 release of the installer.

**Comment by Dark-Thoughts** [comment]
Okay... What's up with the "Failed to start download." crap with this? I've had this happen several times now and I am not sure how to fix it. I don't want to manually download all the mods.

![image](https://github.com/user-attachments/assets/e51b8881-8516-42f3-91f7-25a505feb143)

**Comment by hughesjr** [comment]
> Okay... What's up with the "Failed to start download." crap with this? I've had this happen several times now and I am not sure how to fix it. I don't want to manually download all the mods.
> 
> ![image](https://private-user-images.githubusercontent.com/15324196/369467250-e51b8881-8516-42f3-91f7-25a505feb143.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg4ODU5OTQsIm5iZiI6MTcyODg4NTY5NCwicGF0aCI6Ii8xNTMyNDE5Ni8zNjk0NjcyNTAtZTUxYjg4ODEtODUxNi00MmYzLTkxZjctMjVhNTA1ZmViMTQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE0VDA2MDEzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwNjZjMTFmNjVmMWMxMTAzY2U3NDI5ODlmYmZjZGU0YTY5MWFlN2ZmODE2MGMwM2QzYTQ3ODQxMjM1YzNmNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zIV_EwFz-DSBeZjM5KvsM4SnI81qYqYvZxIzpxP8Zg8)

I have also been getting this from both firefox and chrome when trying to use the Download Manager button on Nexus.
